web/src/App.jsx

```
@@ -92,16 +127,33 @@ const areMessagesEqual = (prev, next) => {
+  const contactInitial = threadAddress ? threadAddress.charAt(0).toUpperCase() : '?';
+
+  return (
+    <div className={`message ${isReceived ? 'received' : 'sent'}`}>
+      {isReceived && (
+        <button
+          className="message-avatar"
+          onClick={() => onAvatarClick && onAvatarClick(threadAddress)}
+          aria-label={`View contact info for ${threadAddress}`}
+          title="View contact info"
+        >
+          {contactInitial}
```

Biome (2.1.2)

[error] Provide an explicit type prop for the button element.

The default type of a button is submit, which causes the submission of a form when placed inside a `form` element. This is likely not the behaviour that you want inside a React application.
Allowed button types are: submit, button or reset

(lint/a11y/useButtonType)